File: crossecore-java/src/java_/NumberLiteralBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class NumberLiteralBase 
extends ExpressionImpl implements NumberLiteral
{
	protected static final java.lang.String TOKENVALUE_EDEFAULT = "";
	private java.lang.String _tokenValue = TOKENVALUE_EDEFAULT;
	public java.lang.String getTokenValue()
	{
		return _tokenValue;
	}
	public void setTokenValue(java.lang.String value){
		
		java.lang.String oldValue = _tokenValue;
		_tokenValue = value;
		if (eNotificationRequired())
			eNotify(new ENotificationImpl(this, Notification.SET, Java_PackageImpl.NUMBERLITERAL_TOKENVALUE, oldValue, value));
		
	}
	
	
	

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.NUMBERLITERAL;
	}
	
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.NUMBERLITERAL_TOKENVALUE:
				return getTokenValue();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.NUMBERLITERAL_TOKENVALUE:
				setTokenValue((java.lang.String) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.NUMBERLITERAL_TOKENVALUE:
				return TOKENVALUE_EDEFAULT == null ? _tokenValue != null : !TOKENVALUE_EDEFAULT.equals(_tokenValue);
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-csharp/Java/Java/Java_/NumberLiteralBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class NumberLiteralBase 
	:ExpressionImpl, NumberLiteral
	{
		private string _tokenValue = "";
		public virtual string tokenValue
		{
		get { 
			return _tokenValue;
		}
		set { _tokenValue = value; }
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.NUMBERLITERAL;
		}
		
		
		
		public override object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.NUMBERLITERAL_TOKENVALUE:
					return tokenValue;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.NUMBERLITERAL_TOKENVALUE:
					tokenValue = (string) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		/*
		public override bool eIsSet(int featureID) {
			switch (featureID) {
				case Java_PackageImpl.NUMBERLITERAL_TOKENVALUE:
					return TOKENVALUE_EDEFAULT == null ? _tokenValue != null : !TOKENVALUE_EDEFAULT.equals(_tokenValue);
			}
			return base.eIsSet(featureID);
		}
		*/
		
		
		
	}
}
